### PR TITLE
Add `--find-links` without `--no-index` rule

### PR DESCRIPTION
**Pull request by stoletheminerals**

### CODE REVIEW

File: assets/semgrep_rules/services/find-links-without-no-index.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+rules:
+  - id: find-links-without-no-index
+    author: Artem Chaikin
+    source: https://github.com/brave/security-action/blob/main/assets/semgrep_rules/services/find-links-without-no-index.yaml
+    references:
+        - https://portswigger.net/daily-swig/dependency-confusion-attack-mounted-via-pypi-repo-exposes-flawed-package-installer-behavior
+        - https://www.bleepingcomputer.com/news/security/pytorch-discloses-malicious-dependency-chain-compromise-over-holidays/
+    confidence: LOW
+    pattern-regex: '^(?!.*--no-index).*--find-links.*$'
```
Comment (bcaller) [inline assets/semgrep_rules/services/find-links-without-no-index.yaml]: Do you need the `.*$` at the end?

```suggestion
    pattern-regex: '^(?!.*--no-index).*--find-links'
```

**Review by bcaller — APPROVED** [review]

**Comment by bcaller** [comment]
Note, this doesn't find `-f`

File: assets/semgrep_rules/services/find-links-without-no-index.yaml (hunk 1 above)
Comment (stoletheminerals) [reply assets/semgrep_rules/services/find-links-without-no-index.yaml]: Changed, thanks!

**Comment by stoletheminerals** [comment]
@bcaller I added your "pip -f" regex. It won't trigger on "pip install -r req.txt" of course, but will trigger on one-liners at least

**Pull request merged**